intelmq/bots/collectors/xmpp/collector.py

```diff
@@ -21,48 +41,87 @@ def init(self):
             self.stop()
 
     def process(self):
-
         if self.xmpp is None:
-            self.xmpp = sleekxmpp.XMPPBot(self.parameters.xmpp_user + '@' + self.parameters.xmpp_server,
-                                          self.parameters.xmpp_password,
-                                          self.send_message,
-                                          self.logger)
+            self.xmpp = XMPPClient(self.parameters.xmpp_user + '@' +
+                                   self.parameters.xmpp_server,
+                                   self.parameters.xmpp_password,
+                                   self.parameters.xmpp_room,
+                                   self.parameters.xmpp_room_nick,
+                                   self.parameters.xmpp_room_password,
+                                   self.logger)
             self.xmpp.connect(reattempt=True)
             self.xmpp.process()
 
-    def killbot(self):
-        self.xmpp.disconnect(wait=True)
-        self.logger.info("Disconnected")
-
+            # Add Handlers and register Plugins
+            self.xmpp.register_plugin('xep_0030')  # Service Discovery
+            self.xmpp.register_plugin('xep_0045')  # Multi-User Chat
+            self.xmpp.add_event_handler("message", self.log_message)
+            self.xmpp.add_event_handler("groupchat_message", self.log_message)
+
+    def stop(self):
+        if self.xmpp:
+            self.xmpp.disconnect()
+            self.logger.info("Disconnected from XMPP")
+
+            super(XMPPCollectorBot, self).stop()
+        else:
+            self.logger.info("There was no XMPPClient I could stop.")
+
+    def log_message(self, msg):
+        self.logger.debug("Received Stanza: %r from %r", msg['body'],
+                          msg['from'])
+        self.logger.info("Stanza received")
```

Comment:
Right, but as we have a debug line anywhere in the line above, let's just remove it?

I think we should also limit the length of the logged stanza to 500 or so.